newIDE/app/public/External/Utils/pathEditor.js

```diff
@@ -0,0 +1,132 @@
+const electron = require('electron');
+const fs = require('fs');
+const remote = electron.remote;
+const {
+    dialog
+} = remote;
+
+const headerStyle = {
+    saveFolderLabel: 'height:27px;color:SlateGrey;float: left;margin-left: 2px;margin-top: 10px; font-size:15px;',
+    nameInput: 'font-family:"Courier New";height:27px;width:90px;color:SlateGrey;float:left;margin-left: 2px;padding:4px;margin-top: 4px;font-size:15px;border: 2px solid #e5cd50;border-radius: 3px;  ',
+    fileExistsLabel: 'height:27px;color:blue;float: left;margin-left: 2px;margin-top: 10px; font-size:15px;',
+
+    saveButton: 'height:27px;float:right;margin-left:2px;margin-right:4px;border: 2px solid DeepSkyBlue;border-radius: 1px;margin-top: 5px;background-color:white;',
+    cancelButton: 'height:27px;float:right;margin-right:2px;border: 2px solid DeepSkyBlue;border-radius: 1px;margin-top: 5px;background-color:white;',
+    setFolderButton: 'height:27px;float:right;margin-left:2px;margin-right:4px;border: 2px solid DeepSkyBlue;border-radius: 1px;margin-top: 5px;background-color:white;',
+};
+
+let saveFolderLabel,
```

Comment:
Same as `saveOptions`: avoid declaring instead declare these variables here. Declare them in in `jsfx-main.js`, and pass them to these functions.

These functions in an util file should be "pure": they should not use any variable that is not passed as a parameter at all. They should just act on whatever is passed to them. This way they are easy to understand and refactor.

Reply:
(and if you have to pass a lot of variables, it's a good sign that you might need an object or multiple functions ;))

Reply:
good thinking, piskel's css is different anyway - so this will help

Reply:
I'm not sure how to move saveOptions and all the dom elements outside of the scope of methods using them. Should I create them as nulls in jsfx-main and pass them as parameters

I made a pull where at least the css is outside

Reply:
I can put the pathEditor and all of its methods in its own class if that makes more sense?

Reply:
ok I have an idea= I can return all of these as an object when creating it

Reply:
> Should I create them as nulls in jsfx-main and pass them as parameters

Yes, create them as null or as a value that makes sense (but null is probably fine).

> I can put the pathEditor and all of its methods in its own class if that makes more sense?

Now that we have a module, it's ok not too have a class :) You can let them as is